fix: resources and services placement

dokolyski opened the pull request:
resources and services moved to data-access lib, imports adjust to it

DominikKalinowski committed:
fix: resources and services placement

resources and services moved to data-access lib, imports adjust to it

kstepien3 commented on libs/tasks/data-access/src/lib/+state/tasks/tasks.reducer.ts:
```diff
@@ -2,17 +2,17 @@ import { createReducer, on, Action } from '@ngrx/store';
-  error?: string | null; // last known error (if any)
-  from?: number;
-  limit?: number;
+  error: string | null; // last known error (if any)
```
probably this should be `HttpErrorResponse`

dokolyski replied:
jak zmienię zaciąganie danych na poprzez http to podmienię.

kstepien3 commented on libs/tasks/data-access/src/lib/+state/tasks/tasks.reducer.ts:
```diff
@@ -2,17 +2,17 @@ import { createReducer, on, Action } from '@ngrx/store';
 export interface TasksState extends EntityState<Task> {
   tasks: Task[];
-  selectedId?: string | number; // which Tasks record has been selected
+  selectedId: string | null; // which Tasks record has been selected
```
I think you don't need null property. Just set an empty string

dokolyski replied:
zmieniłem w końcu na number więc zostawiłem też null

kstepien3 commented on libs/tasks/data-access/src/lib/+state/tasks/tasks.reducer.ts:
```diff
@@ -24,7 +24,11 @@ export const tasksAdapter: EntityAdapter<Task> = createEntityAdapter<Task>();
+  selectedId: null,
+  error: null,
+  from: 0,
+  limit: null
```
remove

DominikKalinowski committed:
feat: ui library added

two components - task list and task form dialog

refactor: feature library added, todo-lists component removed

next libraries structure refactor

fix: data-access lib correctly created, data managing moved from reducer to effects

next libraries structure refactor, task-data service returns observables (not really needed, only for educational purposes :) )

fix: tasks lib

fix: workspace cleaning

feat: TasksFeatureModule loaded in app

...and used in template

fix: omitted TaskEssentials uses removed

fix: unnecessary libraries cleaned up

fix: ng-content instead of ng-container and ng-template

additionally TaskFormDialogComponent import fixed

fix: reducer improvement, tasks-data.service.ts temporary fix

fix: hanging comma in json, correct component export

refactor: ui components moved to the separated libs

ui parent-lib removed

feat: rest api communication, effects fixed

refactor: nested tasks folder removed

fix: task list subscription fixed, formatting

fix: init of task list

refactor: prettier on all files

fix: PR

task library removed, unnecessary rules removed, missing library types added, missing variable types added, unused methods removed

fixup! fix: PR

kstepien3 commented on libs/tasks/data-access/.eslintrc.json:
```diff
@@ -0,0 +1,39 @@
+      "parserOptions": {
+        "project": ["libs/tasks/data-access/tsconfig.*?.json"]
+      },
+      "rules": {
```
You don't need these rules if your module does not type `feature` or `ui`

kstepien3 commented on libs/tasks/feature/src/lib/tasks-feature/tasks-feature.component.html:
```diff
@@ -0,0 +1,25 @@
+  <h1>TODO List</h1>
+  <button mat-raised-button color="primary" (click)='openCreateDialog()'>New task</button>
+</ng-template>
+<todo-workspace-tasks-list [headerTemplate]='todoHeader'
```
Maybe a better way to provide HTML to the component is providing via `ng-conent`

dokolyski replied:
ok

kstepien3 commented on libs/tasks/feature/src/lib/tasks-feature/tasks-feature.component.html:
```diff
@@ -0,0 +1,25 @@
+<ng-template #doneHeader>
+  <h1>DONE List</h1>
+</ng-template>
+<todo-workspace-tasks-list [headerTemplate]='doneHeader'
```
same

kstepien3 commented on nx.json:
```diff
@@ -21,6 +21,9 @@
     "todo": { "tags": [] },
```
add `"type:application"` to tags

kstepien3 commented on nx.json:
```diff
@@ -21,6 +21,9 @@
     "todo": { "tags": [] },
     "todo-data-access": { "tags": [] },
```
add `"type:data-access"` to tags

kstepien3 commented on nx.json:
```diff
@@ -21,6 +21,9 @@
     "todo": { "tags": [] },
     "todo-data-access": { "tags": [] },
     "tasks": { "tags": [] },
```
"task" doesn't tell us what it is.

dokolyski replied:
jasne, to jest biblioteka nadrzędna do feature, data-access i ui, jak to najlepiej byłoby nazwać?

kstepien3 replied:
nie potrzebujesz tego przetrzymywać jako bibliotekę. Możesz zostawić to jako zwykły folder.

kstepien3 commented on nx.json:
```diff
@@ -21,6 +21,9 @@
     "todo-data-access": { "tags": [] },
     "tasks": { "tags": [] },
-    "tasks-feature": { "tags": [] }
+    "tasks-feature": { "tags": [] },
```
`"type:feature"`

kstepien3 commented on angular.json:
```diff
@@ -149,101 +150,126 @@
       }
     },
-    "todo": {
+    "tasks": {
```
remove. You don't need this like a library.

kstepien3 commented on apps/todo-app/src/app/app.module.ts:
```diff
@@ -23,7 +24,7 @@ import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
     ),
     EffectsModule.forRoot([]),
-    !environment.production ? StoreDevtoolsModule.instrument() : []
+    !environment.production ? StoreDevtoolsModule.instrument() : [],
```
I think you have the wrong configuration of eslint, because prettier always add a comma at end line

kstepien3 commented on jest.config.js:
```diff
@@ -1,9 +1,10 @@
     '<rootDir>/apps/todo-app',
-    '<rootDir>/libs/todo',
-    '<rootDir>/libs/todo/data-access',
     '<rootDir>/libs/tasks',
```
remove

kstepien3 commented on libs/tasks/data-access/src/lib/+state/tasks.effects.ts:
```diff
@@ -0,0 +1,98 @@
+      withLatestFrom(
+        this.store$.pipe(map((state) => state[fromTasks.TASKS_FEATURE_KEY]))
+      ),
+      mergeMap(([a, s]) =>
```
you can add type for this

kstepien3 commented on libs/tasks/data-access/src/lib/+state/tasks.reducer.ts:
```diff
@@ -0,0 +1,53 @@
+  error: string | null; // last known error (if any)
+  tasksNumber: number;
+  limit: number | null;
+  from: number | null;
```
Type should be only as a number.

kstepien3 commented on libs/tasks/data-access/src/lib/+state/tasks.reducer.ts:
```diff
@@ -0,0 +1,53 @@
+  tasks: Task[];
+  selectedId: number | null; // which Tasks record has been selected
+  loaded: boolean; // has the Tasks list been loaded
+  error: string | null; // last known error (if any)
```
I think better types to set is `HttpErrorResponse | null`

kstepien3 commented on libs/tasks/data-access/src/lib/resources/models/task.ts:
```diff
@@ -0,0 +1,9 @@
+  completed?: boolean = false;
+  dueDate: Date | null;
+  name: string;
+}
```
change the name of the file to `task.model.ts`

kstepien3 commented on libs/tasks/feature/src/lib/tasks-feature/tasks-feature.component.ts:
```diff
@@ -0,0 +1,31 @@
+  styleUrls: ['./tasks-feature.component.scss'],
+})
+export class TasksFeatureComponent implements OnInit {
+  tasks$: Observable<Task[]>;
```
`tasks$ = this.tasksFacade.tasks$;`
and remove from `ngOnInit`

kstepien3 commented on libs/tasks/feature/src/lib/tasks-feature/tasks-feature.component.ts:
```diff
@@ -0,0 +1,31 @@
+@Component({
+  selector: 'todo-workspace-tasks-feature',
+  templateUrl: './tasks-feature.component.html',
+  styleUrls: ['./tasks-feature.component.scss'],
```
you can sets `ChangeDetection.OnPush` in components for better performance of the application

kstepien3 commented on libs/tasks/src/lib/tasks.module.ts:
```diff
@@ -0,0 +1,7 @@
+@NgModule({
+  imports: [CommonModule],
+})
+export class TasksModule {}
```
you don't need this module

kstepien3 commented on libs/tasks/ui-tasks-list/src/lib/component/tasks-list.component.ts:
```diff
@@ -0,0 +1,26 @@
+
+  constructor() {}
+
+  ngOnInit(): void {}
```
remove

kstepien3 commented on libs/tasks/ui-tasks-list/src/lib/component/tasks-list.component.ts:
```diff
@@ -0,0 +1,26 @@
+export class TasksListComponent implements OnInit, OnChanges {
+  @Input() tasks: Task[];
+
+  constructor() {}
```
remove

kstepien3 commented on libs/tasks/ui-tasks-list/src/lib/component/tasks-list.component.ts:
```diff
@@ -0,0 +1,26 @@
+
+  ngOnInit(): void {}
+
+  ngOnChanges(change) {
```
remove

kstepien3 commented on nx.json:
```diff
@@ -16,11 +16,12 @@
-    "tasks-feature": { "tags": [] }
+    "tasks-feature": { "tags": ["type:feature"] },
+    "tasks-data-access": { "tags": ["type:data-access"] },
+    "tasks-ui-task-form-dialog": { "tags": [] },
```
type ui

kstepien3 commented on nx.json:
```diff
@@ -16,11 +16,12 @@
+    "tasks-feature": { "tags": ["type:feature"] },
+    "tasks-data-access": { "tags": ["type:data-access"] },
+    "tasks-ui-task-form-dialog": { "tags": [] },
+    "tasks-ui-tasks-list": { "tags": [] }
```
same

DominikKalinowski committed:
feat: missing actions added, ui improved

- depConstraints declared
- readme made more concrete
- app component cleaned from automatically generated content
- api key interceptor and proxy added to enable requests to server
(api key declared in environment.ts)
- ui components added: taskList, taskFormDialog and progressSpinner
- snackBar displays occurred errors
- models and interceptors moved to domain directory
- all needed ngrx store actions declared
- reducer, selectors, effects and facade improved
